ingestion/justice_data_source/source.py

```diff
@@ -94,14 +161,65 @@ def _make_chart(self, chart_data) -> MetadataChangeEvent:
         # browse_path = BrowsePathsV2Class(path=["/justice-data/" + "/".join(breadcrumb)])
         # chart_snapshot.aspects.append(browse_path)
 
-        # TODO: propagate ownership from dashboard
+        # add chart ownership
+        owner_urn = builder.make_group_urn(chart_data["owner_email"].split("@")[0])
+        chart_owner = OwnershipClass(
+            owners=[OwnerClass(owner=owner_urn, type="DATAOWNER")]
+        )
+        chart_snapshot.aspects.append(chart_owner)
 
         chart_mce = MetadataChangeEvent(proposedSnapshot=chart_snapshot)
 
         # TODO: add embed url?
 
         return chart_mce
 
+    def _assign_chart_to_domain(self, chart_data) -> MetadataChangeProposalWrapper:
+        """
+        because domain cannot be added via a MetadataChangeEvent we need to use
+        MetadataChangeProposal
+        """
+        domain_urn = builder.make_domain_urn(domain=chart_data["domain"])
+        chart_urn = builder.make_chart_urn(self.platform_name, chart_data["id"])
+
+        mcp = MetadataChangeProposalWrapper(
+            entityType="chart",
+            changeType=ChangeTypeClass.UPSERT,
+            entityUrn=chart_urn,
+            aspect=DomainsClass(domains=[domain_urn]),
+        )
+        return mcp
+
+    def _make_tags_aspect(
+        self, tag_names: list[str] = ["dc_display_in_catalogue"]
+    ) -> GlobalTagsClass:
+        tag_urns = [builder.make_tag_urn(tag=tag_name) for tag_name in tag_names]
+        tag_assocations = [TagAssociationClass(tag_urn) for tag_urn in tag_urns]
+        tags = GlobalTagsClass(tags=tag_assocations)
+        return tags
+
+    def _make_group(self, owner_email) -> MetadataChangeEvent:
+        """
+        these groups will relate to emails on statistcal publication contacts
+        which are not individuals
+        """
+
+        group_urn = builder.make_group_urn(owner_email.split("@")[0])
```

Comment:
Do we want to title case the group name here?

Reply:
I did play around with the idea - I avoided it because sometimes we will sometimes have Acronyms in emails (e.g. `OMSQ-SiC-publications@justice.gov.uk` and this will be all CAPS. Might look weird to have these titled. So i thought just preserve the casing as it's in the email